src/ntlm/mod.rs

```diff
@@ -398,10 +416,20 @@ impl Sspi for Ntlm {
             self.complete_auth_token(&mut [])?;
         }
 
-        SecurityBuffer::find_buffer_mut(message, SecurityBufferType::Token)?; // check if exists
+        let mut encrypted = if let Ok(buffer) = SecurityBuffer::find_buffer_mut(message, SecurityBufferType::Token) {
+            buffer
+        } else {
+            SecurityBuffer::find_buffer_mut(message, SecurityBufferType::Stream)?
+        }
+        .buffer
+        .clone();
         let data = SecurityBuffer::find_buffer_mut(message, SecurityBufferType::Data)?;
+        encrypted.extend_from_slice(&data.buffer);
 
-        *data.buffer.as_mut() = self.recv_sealing_key.as_mut().unwrap().process(data.buffer.as_slice());
+        let signature = encrypted[0..16].to_vec();
+        let enc = encrypted[16..].to_vec();
```

Comment:
Nit: as far as I can see in the rest of the code, there is no need to use `.to_vec()`:
```suggestion
        let signature = &encrypted[0..16];
        let enc = &encrypted[16..];
```
(and remove the `.as_slice()` below)

Reply:
Done